packages/react-scripts/scripts/init.js

```diff
@@ -118,11 +118,37 @@ module.exports = function(
     templateJson = require(templateJsonPath);
   }
 
+  const templatePackage = templateJson.package || {};
+
+  // Keys to ignore in templatePackage
+  const templatePackageBlacklist = [
+    'devDependencies',
+    'peerDependencies',
+    'name',
```

Comment:
I'd suggest we blacklist all keys listed here for now - except for `config` and `engines`:
https://docs.npmjs.com/files/package.json

We can then enable them on a case-by-case basis, unless you can think of some more that might be immediately useful?

Reply:
Nothing jumps out at the moment so I've added them all in to the blacklist, except for `config` and `engines` as requested.

Reply:
Please also remember I asked for a few to be immediately whitelisted here: https://github.com/facebook/create-react-app/pull/8209

I closed my PR in favor of this one but specifically keeping homepage off the blacklist would be great so that we can use homepage: "." which is a webpack recommendation in certain scenarios.

Reply:
Apologies @jimthedev, oversight on my part - I've updated this PR to remove `homepage` from the blacklist so that your PR is covered.

Regarding whitelisting other keys, there is only a blacklist so any keys used that are not in the blacklist (e.g. `browserslist`) will be copied over to the output `package.json`. Here is a [test example](https://github.com/tomvalorsa/cra-template-test/blob/test-06-custom-keys-to-replace-extend/template.json) of a `template.json` file with the `browserslist` field that was run successfully. Instructions to run this are in the initial PR description if you're interested.

Reply:
Thanks much @tomvalorsa. Makes sense. Appreciate you working on this.

Reply:
Thanks for being on top of this @tomvalorsa. I'll try to get someone to review ASAP.